src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,149 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, { useRef } from 'react';
+import { NumberInputIncrement, NumberInputProps } from './types';
+import { merge } from '@utilities/merge';
+import { IconExclamationMarkTriangle16, IconMinus16, IconPlus16 } from '@foundation/Icon';
+import { HelperText } from '@utilities/input';
+import { useMemoizedId } from '@hooks/useMemoizedId';
+import { FOCUS_WITHIN_STYLE } from '@utilities/focusStyle';
+import { Validation, validationClassMap } from '@utilities/validation';
+
+export const NumberInput = ({
+    id: propId,
+    size,
+    error,
+    disabled = false,
+    required,
+    readOnly,
+    errorText,
+    decorator,
+    incrementable,
+    validation = Validation.Default,
+    onChange,
+    onKeyDown,
+    onBlur,
+    onFocus,
+    'data-test-id': dataTestId = 'fondue-number-input',
+    ...props
+}: NumberInputProps) => {
+    const inputElement = useRef<HTMLInputElement | null>(null);
+
+    const handleOnChange = (value: string) => {
+        onChange?.(value);
+    };
+
+    const handleCount = (value: string, type?: NumberInputIncrement, isShift?: boolean) => {
+        const getNewInputValue = () => {
+            const currentValue = Number(value) || 0;
+            const changeValue = isShift ? 10 : 1;
+            switch (type) {
+                case NumberInputIncrement.DECREMENT:
+                    return currentValue - changeValue;
+                case NumberInputIncrement.INCREMENT:
+                    return currentValue + changeValue;
+                default:
+                    return currentValue;
+            }
+        };
+        if (inputElement.current) {
+            const newValue = getNewInputValue();
+            inputElement.current.value = newValue.toString();
+            handleOnChange(inputElement.current.value);
+        }
+    };
+
+    return (
+        <div data-test-id={dataTestId}>
+            <div
+                className={merge([
+                    'tw-flex tw-items-center tw-h-9 tw-gap-2 tw-px-3 tw-transition tw-text-s tw-font-sans tw-relative tw-bg-white dark:tw-bg-transparent tw-border tw-rounded tw-line-strong hover:tw-line-x-strong focus-within:tw-line-xx-strong',
+                    FOCUS_WITHIN_STYLE,
+                ])}
+            >
+                {decorator ? (
+                    <div
+                        className={merge([
+                            'tw-flex tw-items-center tw-justify-center tw-pl-1',
+                            disabled || readOnly ? 'tw-text-black-60' : 'tw-text-black-80',
+                        ])}
+                        data-test-id={`${dataTestId}-decorator`}
+                    >
+                        {decorator}
+                    </div>
+                ) : null}
+                <input
```

Comment:
Right now there is no behavior on mouse hold, should we add an increment 
```
hold 0-2s: +1
hold 2-5s: +5
hold 5-10s: +10
```
For examplem it could be a simple props like `stepHoldInterval` with default like
```tsx
stepHoldInterval={(t) => fib(t)}
```

Reply:
Agree.  I though about this as well that it could be a nice feature to have.  Also with the `shiftDown` increment should we enable a `stepInterval` prop?

Reply:
Should work on both mouseHold and with Keyboard navigation using `Shift`
You can tale a look at the `Slider` component: https://beta-fondue-components.frontify.com/?path=/story/components-slider--basic-usage